internal/app/accrual/accrual.go

```diff
@@ -0,0 +1,63 @@
+package accrual
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/mihailtudos/gophermart/internal/domain"
+	"github.com/mihailtudos/gophermart/internal/logger"
+)
+
+const httpClientTimeout = time.Minute
+
+type AccrualClient interface {
```

Comment:
You are right, I totally agree with you. This looks like interface polluation. 

Assumming, I can swap the HTTP client (Accrual Client) with an implementation  based on another HTTP client such as Resty it's worth using an interface but it's place as you said should be where it's going to be used. In this case, it would be the service package itself. 

```go
type Services struct {
	UserService   *UserService
	TokenManager  TokenManager
	AccrualClient AccrualClient
}
```

Reply:
To be honest with you, although I went through the resources you gave me but I still haven't managed to use interfaces "the Go way".

Reply:
You could read the book "101 mistakes in Go" about "Interface Pollution" particularly page 24. Here's the beginning:" It’s fairly common to see interfaces being overused in Go projects. Perhaps the developer’s background was C# or Java, and they found it natural to create interfaces before concrete types. However, this isn’t how things should work in Go."

Reply:
"As we discussed, interfaces are made to create abstractions. And the main caveat when programming meets abstractions is remembering that abstractions should be dis- covered, not created. What does this mean? It means we shouldn’t start creating abstrac- tions in our code if there is no immediate reason to do so. We shouldn’t design with interfaces but wait for a concrete need. Said differently, we should create an interface when we need it, not when we foresee that we could need it"

Reply:
these steps will help you:
1. create a realisation in package A
2. create a function-constructor that will return pointer to the realisation
3. write an interface in package B that will describe only behaviour that the package is waiting for
4. write the constructor of package B that will require the interface
Done!

Reply:
Yeah, it makes sense. I think where I went wrong was that I should have created multiple services for example User, Order, Balance, Auth, etc. But I only created one User Service and the interface it implements is big which I was not happy with. 

But I realised I could split the interface into smaller interfaces and later when I decided to have an Order Service it would implement that specific small interface.

Also, I am not sure how to get rid of duplicate interface definitions as I have 3 large interconnected layers and the repo needs a simpler interface as the service as well as the handlers. Here is where I am getting confused.